main.go

```diff
@@ -169,6 +179,61 @@ func main() {
 	os.Exit(evaluateOutcome(summary, test.Config.Expected)) // Returns success on all tests to OS; this allows for test scripting.
 }
 
+/* parse all input args of the form `parameter=value`, e.g. `X=5` */
+func parseArgs(args []string) map[string]string {
+	parsedArgs := make(map[string]string)
+	for _, arg := range args {
+		splitArgs := strings.Split(arg, "=")
+		if len(splitArgs) != 2 {
+			fmt.Println("Bad assignment: " + arg)
+			os.Exit(1)
+		}
+
+		param, val := splitArgs[0], splitArgs[1]
+		parsedArgs[param] = val
+	}
+	return parsedArgs
+}
+
+/* resolve all parameters in test file */
+func replaceAllParams(fileData []byte, params map[string]string) []byte {
+	// replace all given parameters with their specified values
+	fileData = replaceInputParams(fileData, params)
+	// replace all other parameters with their default values
+	fileData = replaceDefaultParams(fileData)
+
+	return fileData
+}
+
+func compileParamRegex(param string) *regexp.Regexp {
+	// matches strings of the form `%{parameter}` and `%{parameter, value}`
+	paramRegex, _ := regexp.Compile("%\\{\\s*" + param + "\\s*(?:,\\s*.*?\\s*)?\\}")
+	return paramRegex
+}
+
+/* resolve all param names (keys in `params` map) to their resp. values */
+func replaceInputParams(fileData []byte, params map[string]string) []byte {
+	for param, val := range params {
+		paramRegex := compileParamRegex(param)
+		fileData = paramRegex.ReplaceAll(fileData, []byte(val))
```

Comment:
This is not the best way to do replacement as it offers little information of what actually happened and what you left behind unsubstituted.

Normally, you'd:

1. Scan for token (your token being `{{ something }}`
2. Find value for `something`. Tell the user if `something` does not have a value, is malformed etc.
3. Replace the token

I have looked and I have not found a library in go that allows to tokenize/replace easily. A combination of `regexp.FindAll()` and `bytes.Replace()` might make it.

With the first you find all occurences of `{{ variable }}`. Then for each of them you check that you have a value (or fail loudly otherwise). When you do, you call bytes.Replace() on the yml to substitute that occurence with the value. You can keep track of already replaced expressions so you don't do it twice.